### PR TITLE
Updates aws-sdk from v2 to v3

### PR DESCRIPTION
**Pull request by patrickberkeley**
Main changes:

* Use the new `@aws-sdk/lib-storage` and `@aws-sdk/client-s3` packages instead of the old `aws-sdk` package
* [Global `AWS.config` is no longer supported](https://github.com/aws/aws-sdk-js-v3#configuration). Push this down into the configuration for the client.
* Refactor from promises chaining [to async/await as recommended](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-s3/index.html#asyncawait).

### CODE REVIEW

**Comment by patrickberkeley** [comment]
[Migrate our docker-pgdump-s3 image to AWS SDK v3](https://trello.com/c/PhKah5HV/3936-migrate-our-docker-pgdump-s3-image-to-aws-sdk-v3)

**Review by beerlington — DISMISSED** [review]

File: lib/upload-s3.js [hunk 1]
```diff
@@ -1,49 +1,57 @@
...
-module.exports = function (stream, config, key) {
+module.exports = async (stream, { S3_BUCKET, S3_REGION, S3_STORAGE_CLASS, S3_URL }, Key) => {
     if (!stream || typeof stream.on !== 'function') {
         throw new Error('invalid stream provided')
     }
-    return new Promise((resolve, reject) => {
-        const timer = new Timer();
+
+    const passThrough = new PassThrough()
+    pipeline(stream, passThrough, () => {})
```
Comment (patrickberkeley) [inline lib/upload-s3.js:12]: Apparently the new sdk expects a Stream, so we create one here (more info: https://github.com/aws/aws-sdk-js-v3/issues/2522#issuecomment-939457957)

File: lib/upload-s3.js [hunk 2]
```diff
@@ -1,49 +1,57 @@
-const AWS = require('aws-sdk')
+const { Upload } = require('@aws-sdk/lib-storage')
+const { S3, S3Client } = require('@aws-sdk/client-s3')
 const Timer = require('./timer')
+const { PassThrough, pipeline } = require('stream')
 
-// configure AWS to log to stdout
-AWS.config.update({
```
Comment (patrickberkeley) [inline lib/upload-s3.js:5]: Logging seems to work OK without configuring anything e.g., 

```
}
{
  loaded: [33m9500098560[39m,
  total: [90mundefined[39m,
  part: [33m1812[39m,
  Key: [32m'postgres-2023-03-16.backup'[39m,
  Bucket: [32m'patrick-docker-pgdump-s3-test'[39m
}
{
  loaded: [33m9505341440[39m,
  total: [90mundefined[39m,
  part: [33m1813[39m,
  Key: [32m'postgres-2023-03-16.backup'[39m,
  Bucket: [32m'patrick-docker-pgdump-s3-test'[39m
}
{
  loaded: [33m9510584320[39m,
  total: [90mundefined[39m,
  part: [33m1814[39m,
  Key: [32m'postgres-2023-03-16.backup'[39m,
  Bucket: [32m'patrick-docker-pgdump-s3-test'[39m
}
{
  loaded: [33m9515827200[39m,
  total: [90mundefined[39m,
  part: [33m1815[39m,
  Key: [32m'postgres-2023-03-16.backup'[39m,
  Bucket: [32m'patrick-docker-pgdump-s3-test'[39m
}
{
  loaded: [33m9521070080[39m,
  total: [90mundefined[39m,
  part: [33m1816[39m,
  Key: [32m'postgres-2023-03-16.backup'[39m,
  Bucket: [32m'patrick-docker-pgdump-s3-test'[39m
}
{
  loaded: [33m9526312960[39m,
  total: [90mundefined[39m,
  part: [33m1817[39m,
  Key: [32m'postgres-2023-03-16.backup'[39m,
  Bucket: [32m'patrick-docker-pgdump-s3-test'[39m
}
{
  loaded: [33m9531555840[39m,
  total: [90mundefined[39m,
  part: [33m1818[39m,
  Key: [32m'postgres-2023-03-16.backup'[39m,
  Bucket: [32m'patrick-docker-pgdump-s3-test'[39m
}
{
  loaded: [33m9536798720[39m,
  total: [90mundefined[39m,
  part: [33m1819[39m,
  Key: [32m'postgres-2023-03-16.backup'[39m,
  Bucket: [32m'patrick-docker-pgdump-s3-test'[39m
}
{
  loaded: [33m9542041600[39m,
  total: [90mundefined[39m,
  part: [33m1820[39m,
  Key: [32m'postgres-2023-03-16.backup'[39m,
  Bucket: [32m'patrick-docker-pgdump-s3-test'[39m
}
{
  loaded: [33m9547284480[39m,
  total: [90mundefined[39m,
  part: [33m1821[39m,
  Key: [32m'postgres-2023-03-16.backup'[39m,
  Bucket: [32m'patrick-docker-pgdump-s3-test'[39m
}
{
  loaded: [33m9552527360[39m,
  total: [90mundefined[39m,
  part: [33m1822[39m,
  Key: [32m'postgres-2023-03-16.backup'[39m,
  Bucket: [32m'patrick-docker-pgdump-s3-test'[39m
}
{
  loaded: [33m9557770240[39m,
  total: [90mundefined[39m,
  part: [33m1823[39m,
  Key: [32m'postgres-2023-03-16.backup'[39m,
  Bucket: [32m'patrick-docker-pgdump-s3-test'[39m
}
{
  loaded: [33m9563013120[39m,
  total: [90mundefined[39m,
  part: [33m1824[39m,
  Key: [32m'postgres-2023-03-16.backup'[39m,
  Bucket: [32m'patrick-docker-pgdump-s3-test'[39m
}
{
  loaded: [33m9566988964[39m,
  total: [90mundefined[39m,
  part: [33m1825[39m,
  Key: [32m'postgres-2023-03-16.backup'[39m,
  Bucket: [32m'patrick-docker-pgdump-s3-test'[39m
}
SUCCESS: The backup was uploaded to https://patrick-docker-pgdump-s3-test.s3.us-east-1.amazonaws.com/postgres-2023-03-16.backup in 00:14:00:796
```

**Comment by patrickberkeley** [comment]
@beerlington sorry for the churn, but I've changed this a decent amount to get it to work since you first reviewed it.

**Comment by patrickberkeley** [comment]
I've replaced the container app for doing backups on crunchy with what's contained in this PR and published to https://hub.docker.com/layers/seated/docker-pgdump-s3/1.3.0-pg-15.2/images/sha256-f743bd224bee1d6a973a5cdc0cc0519d7e5135bded6da1f55b919f07a64e7587?context=explore using the command listed in the stone readme:

```
SELECT run_container('-dt -e PGPASSWORD=<pw> -e PGDATABASE=postgres -e PGUSER=postgres -e PGHOST=<host> -e S3_BUCKET=seated-production-db-backups -e S3_REGION=us-east-1 -e AWS_ACCESS_KEY_ID=<key> -e AWS_SECRET_ACCESS_KEY=<secret> -e PG_PARAMS="-Fc" -e BACKUP_EXTENSION=backup -e S3_STORAGE_CLASS=STANDARD -e CRON_SCHEDULE="0 1 * * *" -e S3_URL="seated-production-db-backups.us-east-1.amazonaws.com" seated/docker-pgdump-s3:1.3.0-pg-15.2');
```

**Review by beerlington — APPROVED** [review]

**Pull request merged**